scipy/optimize/_chandrupatla.py

```diff
@@ -33,6 +33,7 @@ def _chandrupatla(func, a, b, *, args=(), xatol=_xtol, xrtol=_rtol,
         See Notes for details.
     maxiter : int, optional
         The maximum number of iterations of the algorithm to perform.
+        Default is 2048.
```

Comment:
Maybe make the maximum number of iterations depend on the `dtype`?  The default could be `None` and then `maxiter` could be calculated as

```python
maxiter = 2**(np.finfo(dtype).nexp - 1)*2
```

But then you'd need to document why `maxiter` is calculated this way.

Reply:
Yeah that's a good idea. How about just documenting that "The default is the maximum number of bisections of the (normal) floating point numbers."? We could append "of the dtype" if that needs to be explicit.

Reply:
That sounds good to me. Either with or without appending "of the input dtype".

Reply:
Done.